protocol/xmrmaker/swap_state.go

```diff
@@ -448,14 +448,25 @@ func (s *swapState) checkContract(txHash ethcommon.Hash) error {
 		return fmt.Errorf("failed to get receipt for New transaction: %w", err)
 	}
 
+	if receipt.Status == 0 {
+		// swap transaction reverted
+		return errLockTxReverted
+	}
+
 	// check that New log was emitted
 	if len(receipt.Logs) == 0 {
 		return errCannotFindNewLog
 	}
 
-	event, err := s.Contract().ParseNew(*receipt.Logs[0])
+	var event *swapfactory.SwapFactoryNew
+	for _, log := range receipt.Logs {
+		event, err = s.Contract().ParseNew(*log)
+		if err == nil {
+			break
+		}
+	}
```

Comment:
is there a case where there would be multiple logs in one receipt? I don't think there would be?

Reply:
Yes, when a token transfer happens, it emits a Transfer event. And also, if the claim is made as an internal contract call, like from within a different smart contract, it may also have other log events.